pkg/controller/datadogagentdeployment/utils.go

```diff
@@ -1018,10 +1019,74 @@ func getPodAffinity(affinity *corev1.Affinity, labelValue string) *corev1.Affini
 	}
 }
 
+func updateDaemonSetStatus(ds *appsv1.DaemonSet, dsStatus *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus, updateTime *metav1.Time) *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus {
+	if dsStatus == nil {
+		dsStatus = &datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus{}
+	}
+	if ds == nil {
+		dsStatus.State = string(datadoghqv1alpha1.DatadogAgentDeploymentStateFailed)
+		return dsStatus
+	}
+	if updateTime != nil {
+		dsStatus.LastUpdate = updateTime
+	}
+
+	dsStatus.CurrentHash = getHashAnnotation(ds.Annotations)
+	dsStatus.Desired = ds.Status.DesiredNumberScheduled
+	dsStatus.Current = ds.Status.CurrentNumberScheduled
+	dsStatus.Ready = ds.Status.NumberReady
+	dsStatus.Available = ds.Status.NumberAvailable
+	dsStatus.UpToDate = ds.Status.UpdatedNumberScheduled
+
+	var deploymentState datadoghqv1alpha1.DatadogAgentDeploymentState
+	if dsStatus.UpToDate != dsStatus.Desired {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateUpdating
+	} else if dsStatus.Ready == 0 {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateProgressing
+	} else {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateRunning
+	}
+
+	dsStatus.State = fmt.Sprintf("%v (%d/%d/%d)", deploymentState, dsStatus.Desired, dsStatus.Ready, dsStatus.UpToDate)
+	return dsStatus
+}
+
+func updateExtendedDaemonSetStatus(eds *edsdatadoghqv1alpha1.ExtendedDaemonSet, dsStatus *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus, updateTime *metav1.Time) *datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus {
+	if dsStatus == nil {
+		dsStatus = &datadoghqv1alpha1.DatadogAgentDeploymentAgentStatus{}
+	}
+	if updateTime != nil {
+		dsStatus.LastUpdate = updateTime
+	}
+	dsStatus.CurrentHash = getHashAnnotation(eds.Annotations)
+	dsStatus.Desired = eds.Status.Desired
+	dsStatus.Current = eds.Status.Current
+	dsStatus.Ready = eds.Status.Ready
+	dsStatus.Available = eds.Status.Available
+	dsStatus.UpToDate = eds.Status.UpToDate
+
+	var deploymentState datadoghqv1alpha1.DatadogAgentDeploymentState
+	if dsStatus.UpToDate != dsStatus.Desired {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateUpdating
+	} else if dsStatus.Ready == 0 {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateProgressing
+	} else {
+		deploymentState = datadoghqv1alpha1.DatadogAgentDeploymentStateRunning
```

Comment:
what about using a `switch {}` instead of the `if else if else` ?

I think this logic can be added directly to the EDS controller. like this you just need to get the state value. 

Also here we are loosing the information when the EDS is in a Canary state. Is this wanted? Do you think it is not a relevant information?

Reply:
Fine with a `switch{}`. We could add it to the EDS, but then we need to keep the logic for normal DS, so not sure it's worth it?

A `Canary` state existed but was never set in the code, so I am not sure we are loosing something :D. Currently it will be flagged as `Updating`, though if there is a specific way to detect `Canary` state in EDS, I can add it there.

Reply:
About the "Canary" state: the intension was here :) You can consider that was a bug.

To detect thea canary state you have in the [EDS.status.canary](https://github.com/DataDog/extendeddaemonset/blob/master/pkg/apis/datadoghq/v1alpha1/extendeddaemonset_types.go#L103) section that appear when a canary is ongoing. Else the [EDS.status.state](https://github.com/DataDog/extendeddaemonset/blob/master/pkg/apis/datadoghq/v1alpha1/extendeddaemonset_types.go#L101) should reflect it too

Reply:
Added!